feature-libs/product-configurator/rulebased/cpq/cpq-configurator-rest.service.ts

```diff
@@ -0,0 +1,235 @@
+   * and read this default configuration from the CPQ system.
+   */
+  createConfiguration(
+    productSystemId: string
+  ): Observable<Configurator.Configuration> {
+    return this.callConfigurationInit(productSystemId).pipe(
+      switchMap((configCreatedResponse) => {
+        return this.callConfigurationDisplay(
+          configCreatedResponse.configurationId
+        ).pipe(
+          this.converterService.pipeable(CPQ_CONFIGURATOR_NORMALIZER),
+          map((resultConfiguration) => {
+            return {
+              ...resultConfiguration,
+              configId: configCreatedResponse.configurationId,
+            };
+          })
+        );
+      })
+    );
+  }
+
+  readConfiguration(
+    configId: string,
+    tabId?: string
+  ): Observable<Configurator.Configuration> {
+    return this.callConfigurationDisplay(configId, tabId).pipe(
+      this.converterService.pipeable(CPQ_CONFIGURATOR_NORMALIZER),
+      map((resultConfiguration) => {
+        return {
+          ...resultConfiguration,
+          configId: configId,
+        };
+      })
+    );
+  }
+
+  readConfigurationOverview(
+    configId: string
+  ): Observable<Configurator.Overview> {
+    return this.getConfigurationWithAllTabsAndAttribues(configId).pipe(
+      this.converterService.pipeable(CPQ_CONFIGURATOR_OVERVIEW_NORMALIZER),
+      map((resultConfiguration) => {
+        return {
+          ...resultConfiguration,
+          configId: configId,
+        };
+      })
+    );
+  }
+
+  /**
+   * This method is actually a workarround until CPQ provides and API to fetch
+   * all selected attributes / attribue values grouped by all tabs.
+   * It will fire a request for each tab to collect all required data.
+   */
+  protected getConfigurationWithAllTabsAndAttribues(
+    configId: string
+  ): Observable<Cpq.Configuration> {
+    return this.callConfigurationDisplay(configId).pipe(
+      switchMap((currentTab) => {
+        const tabRequests: Observable<Cpq.Configuration>[] = [];
+        if (currentTab.tabs && currentTab.tabs.length > 0) {
+          // prepare requests for remaining tabs
+          currentTab.tabs.forEach((tab) => {
+            if (tab.isSelected) {
+              // details of the currently selected tab are already fetched
+              tabRequests.push(of(currentTab));
+            } else {
+              tabRequests.push(
+                this.callConfigurationDisplay(configId, tab.id.toString())
+              );
+            }
+          });
+        } else {
+          // tabs are not defined in model, general tab is used
+          tabRequests.push(of(currentTab));
+        }
+        // fire requests for remaining tabs and wait until all are finished
+        return forkJoin(tabRequests);
+      }),
+      map(this.mergeTabResults)
+    );
+  }
+
+  protected mergeTabResults(
+    tabReqResultList: Cpq.Configuration[]
+  ): Cpq.Configuration {
+    const ovConfig = {
```

Comment:
What does `ov` mean? Should we consider rename or adding a comment in the code?

Reply:
Pick!

Reply:
Done!